### PR TITLE
Rory/podium

### PR DESCRIPTION
**Pull request by roryvision**
### Description

(_What have been done/changed_)

- Set up the front end for podium judging portal with the initial homepage and showcase slide room, both with ranking capabilities.
- Bug that needs to be reviewed: The spotlighted project in the slide room, when dragged into the ranking bar, will have a NULL value. I'm not sure what is causing this since visually, still looks fine so the project information is probably being passed through correctly. This is a problem because the project will now not be able to be ranked against.

### To Do

- [x] Call judges endpoint instead of vertical (also hardcoded right now, it should get the judge ID from their session)
- [x] Currently stores the ranking in the app instead of making a POST request, to avoid messing up the database. Should make some adjustments to the current examples to test properly.
- [ ] Finish On Hold sections (currently makes multiple copies if a project is dragged in multiple times, etc.) after fixing spotlight bug.
- [ ] Improve loading visual.
- [ ] Product designer still needs to finish the header.
- [ ] Allow users to expand the spotlight descriptions.
- [x] Not a big UX problem, but would be great if there was animation as it resorted (during drag instead of just on end), it should come automatically with SortableContext in dnd-kit but something seems to be stopping it from triggering.

### Testing plan

(_How will we test the changes requested on this PR?_)

- [ ] Begin on index, where you can drag projects around. If a project is dragged into rank 4, it will push the current 4 and all those below down. Projects cannot be dragged onto unranked projects (intentional).
- [ ] Click on a project name to enter the slide room. Drag the spotlight into the bottom bar to rank it (cannot be dragged onto unranked, intentional). Click on a project to bring it up as the spotlight. You cannot rank from the bar, only from spotlight—this is intentional for UX reasons. There is a bug (see notes) on this page, so it might not work as expected.

### CODE REVIEW

**Review by VictiniX888 — COMMENTED** [review]
Some review comments! I'll add my commit with the user integration and then I'll merge the PR  (just so we can get a preview out sooner). Feel free to address the review comments in a separate PR along with other additions

File: apps/podium/utils/getProjects.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import axios from 'axios';
+
+// TODO: replace hardcode with vertical id
+// TODO: fetch from session judge id
+
+const HIBISCUS_API_URL = process.env.NEXT_PUBLIC_HIBISCUS_API_URL;
```
[the remark this answers is not in the document]
Comment (VictiniX888) [reply apps/podium/utils/getProjects.ts:6]: Also, I would strongly recommend naming the env var something specific to the judging portal e.g. NEXT_PUBLIC_HIBISCUS_PODIUM_API_URL

File: apps/podium/utils/getProjects.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    const jsonString = JSON.stringify(response.data);
+    const jsonData = JSON.parse(jsonString);
+
+    console.log(jsonData);
+
+    return jsonData.projects;
+  } catch (error) {
+    console.error('Error: ', error);
+  }
+}
```
[the remark this answers is not in the document]
Comment (roryvision) [reply apps/podium/utils/getProjects.ts:27]: Should be fine, it'll be useless after user integration anyways.

**Comment by VictiniX888** [comment]
@roryvision if you could go ahead and make a new PR from the rory/podium-fixed branch

**Pull request closed without merging**